app/uk/gov/hmrc/gform/sharedmodel/formtemplate/Page.scala

```diff
@@ -39,8 +40,35 @@ case class Page(
     case IsRevealingChoice(revealingChoice) => revealingChoice.options.toList.flatMap(_.revealingFields)
     case otherwise                          => otherwise :: Nil
   }
+  val allFormComponentIds: List[FormComponentId] = (fields ++ fields
+    .flatMap {
+      case IsGroup(group)                     => group.fields
+      case IsRevealingChoice(revealingChoice) => revealingChoice.options.toList.flatMap(_.revealingFields)
+      case otherwise                          => Nil
+    })
+    .map(_.id)
+
+  val numericFields: List[FormComponentId] = allFormComponents.filter(_.isNumeric).map(_.id)
 }
 
 object Page {
   implicit val pageFormat: OFormat[Page] = Json.format[Page]
+
+  implicit val leafExprs: LeafExpr[Page] = new LeafExpr[Page] {
+    def exprs(path: TemplatePath, t: Page): List[ExprWithPath] =
```

Comment:
That would break consistency, since all `implicit` values of `LeafExpr` are called `leafExprs`

Reply:
Ok, just makes the variable not so intutive, need to drill down to find out. Does it mean

1. list of leaf exprs for the page
2. a generator class for the page

It may be just me, so feel free to ignore

Reply:
Don't worry Scala3 has your concern covered.... No need to name something you don't want to name there.